src/effects/backends/effectsbackendmanager.cpp

```diff
@@ -0,0 +1,115 @@
+#include "effects/backends/effectsbackendmanager.h"
+
+#include "control/controlobject.h"
+#include "effects/backends/builtin/builtinbackend.h"
+#include "effects/backends/effectprocessor.h"
+#ifdef __LILV__
+#include "effects/backends/lv2/lv2backend.h"
+#endif
+#include "effects/presets/effectpreset.h"
+
+EffectsBackendManager::EffectsBackendManager() {
+    m_pNumEffectsAvailable = std::make_unique<ControlObject>(
+            ConfigKey("[Master]", "num_effectsavailable"));
+    m_pNumEffectsAvailable->setReadOnly();
+
+    addBackend(EffectsBackendPointer(new BuiltInBackend()));
+#ifdef __LILV__
+    addBackend(EffectsBackendPointer(new LV2Backend()));
+#endif
+}
+
+void EffectsBackendManager::addBackend(EffectsBackendPointer pBackend) {
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return;
+    }
+
+    m_effectsBackends.insert(pBackend->getType(), pBackend);
+
+    QList<QString> backendEffects = pBackend->getEffectIds();
+    for (const QString& effectId : backendEffects) {
+        m_manifests.append(pBackend->getManifest(effectId));
+    }
+
+    m_pNumEffectsAvailable->forceSet(m_manifests.size());
+
+    std::sort(m_manifests.begin(),
+            m_manifests.end(),
+            EffectManifest::alphabetize);
+}
+
+const QList<EffectManifestPointer> EffectsBackendManager::getManifestsForBackend(
+        EffectBackendType backendType) const {
+    auto pBackend = m_effectsBackends.value(backendType);
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return QList<EffectManifestPointer>();
+    }
+    auto list = pBackend->getManifests();
+    std::sort(list.begin(), list.end(), EffectManifest::alphabetize);
+    return list;
+}
+
+EffectManifestPointer EffectsBackendManager::getManifestFromUniqueId(
+        const QString& uid) const {
+    if (kEffectDebugOutput) {
+        //qDebug() << "EffectsBackendManager::getManifestFromUniqueId" << uid;
+    }
```

Comment:
I guess this is only for testing purposes? Let's try to get the logging branch merged to make stuff like this unnecessary.

Reply:
Yes, this can be improved when the logging changes are in master.